### PR TITLE
[dash0] deploy new releases automatically

### PR DESCRIPTION
**Pull request by basti1302**

### CODE REVIEW

**Comment by basti1302** [comment]
I yoinked most of this from https://github.com/dash0hq/dash0/blob/main/.github/actions/deploy-development/action.yaml and adapted it for the otel-demo.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -13,3 +13,17 @@ jobs:
...
+  deploy_release:
+    name: deploy to https://otel-demo.eu-west-1.aws.dash0-demo.com/
+    needs:
+      - build_and_push_images
+    runs-on: ubuntu-latest
+    # do not run multiple deployment jobs concurrently
+    concurrency: deploy
+    steps:
+      - name: deploy release
+        uses: .github/actions/deploy-development
```
Comment (marcelbirkner) [inline .github/workflows/release.yml]: Typo: I believe the path should be "deploy-demo" or you rename the directory

Comment (basti1302) [reply .github/workflows/release.yml]: right!

Comment (basti1302) [reply .github/workflows/release.yml]: fixed

**Review by marcelbirkner — APPROVED** [review]

**Pull request merged**